src/main/java/sorald/sonar/RuleVerifier.java

```diff
@@ -54,6 +83,78 @@ public static void verifyHasIssue(List<String> filesToScan, JavaFileScanner chec
      */
     @SuppressWarnings("UnstableApiUsage")
     public static void verifyNoIssue(String filename, JavaFileScanner check) {
-        MultipleFilesJavaCheckVerifier.verifyNoIssue(Arrays.asList(filename), check);
+        JavaCheckVerifier.newVerifier().onFile(filename).withCheck(check).verifyNoIssues();
+    }
+
+    private static InputFile toInputFile(File baseDir, String filename) {
+        // must append a separator to the basedir string as Sonar appends the filenames directly to
+        // it
+        final String baseDirStr = baseDir.toString() + File.separator;
+        try {
+            return new TestInputFileBuilder(baseDirStr, filename)
+                    .setContents(new String(Files.readAllBytes(Paths.get(filename)), UTF_8))
+                    .setCharset(UTF_8)
+                    .setLanguage("java")
+                    .build();
+        } catch (IOException e) {
+            throw new RuntimeException("failed to read file " + filename);
+        }
+    }
+
+    @SuppressWarnings("UnstableApiUsage")
+    private static JavaAstScanner createAstScanner(
+            SonarComponents sonarComponents, List<JavaFileScanner> checks) {
+        JavaAstScanner scanner = new JavaAstScanner(sonarComponents);
+        VisitorsBridge visitorsBridge =
+                new VisitorsBridge(
+                        checks,
+                        // TODO set the classpath to something reasonable
+                        Collections.emptyList(),
```

Comment:
This argument sets the classpath for the VisitorsBridge. It's later supplied to the parser, so I'm assuming that it's used to resolve types etc (but I haven't checked inside the parser). We will probably want to set it to something reasonable in the context. For example, when analyzing a project that uses maven, the full classpath can be generated with `mvn dependency:build-classpath`.